adoc/code/specialization_id.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2011-2021 The Khronos Group, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <sycl/sycl.hpp>
+using namespace sycl; // (optional) avoids need for "sycl::" before SYCL names
+
+struct Mixed {int i; float f;};
+
+constexpr specialization_id<int> a{1};            // OK
+static constexpr specialization_id<int> b{2};     // OK
+specialization_id<int> c{3};                      // ERROR: not constexpr
```

Comment:
When we say "ERROR", does that mean we expect the compiler to diagnose it?

Reply:
I only want to convey that the declaration on this line in non-conformant.  The spec did not say before that the compiler was required to diagnose any errors with declarations of `specialization_id` variables, and I did not want to add that requirement now.

I see that the spec has an existing example which uses the word "illegal" to describe non-conformant code like this, so I think I should change these lines like so:

```
specialization_id<int> c{3};                      // ILLEGAL: not constexpr
```